src/components/work/overview/covercarousel/CoverCarousel.js

```diff
@@ -156,29 +152,30 @@ export function CoverCarousel({
   );
 }
 
-export default function Wrap({ selectedPids, workTitles }) {
+export default function Wrap({ allPids, selectedPids, workTitles }) {
   const { data: manifestationsData, isLoading: manifestationsIsLoading } =
     useData(
       selectedPids?.length > 0 &&
         manifestationFragments.editionManifestations({
-          pid: selectedPids,
+          pid: allPids,
         })
     );
 
-  const { manifestationsWithCover, materialType } = useMemo(() => {
+  const { manifestationsWithCover } = useMemo(() => {
     return getManifestationsWithCorrectCover(
-      manifestationsData?.manifestations
+      manifestationsData?.manifestations?.filter((manifestation) =>
+        selectedPids?.includes(manifestation?.pid)
+      )
     );
-  }, [manifestationsData?.manifestations]);
+  }, [manifestationsData?.manifestations, selectedPids]);
 
-  if (manifestationsIsLoading) {
-    <Skeleton className={styles.carousel_skeleton} />;
+  if (!manifestationsData || manifestationsIsLoading) {
```

Comment:
den kan vist gå i uendelig skeleton .. er det ikke nok med loading ?

Reply:
Jeps, det er muligvis rigtigt, så for en god ordens skyld fjerner jeg det